### PR TITLE
chore: Use SKR Module List as single source of Truth

### PR DESCRIPTION
**Pull request by jeremyharisch**
**Description**

Changes proposed in this pull request:

- Implements https://github.com/kyma-project/lifecycle-manager/issues/756
**Related issue(s)**

### CODE REVIEW

**Comment by jeremyharisch** [comment]
Lets wait for https://github.com/kyma-project/lifecycle-manager/pull/834 to be merged, since the tests introduced in https://github.com/kyma-project/lifecycle-manager/pull/834 needs to be adapted with this change in the current PR.

**Review by ruanxin — CHANGES_REQUESTED** [review]

**Review by ruanxin — CHANGES_REQUESTED** [review]
implmentation looks good, check my comments for test.

File: controllers/control-plane/kyma_remote_sync_test.go [hunk 1]
```diff
@@ -243,6 +235,58 @@ var _ = Describe("Kyma sync into Remote Cluster", Ordered, func() {
...
+		Eventually(UpdateManifestState, Timeout, Interval).
+			WithArguments(ctx, controlPlaneClient, kyma, moduleInKCP, v1beta2.StateReady).
+			Should(Succeed())
+
+		By("Delete default module from remote Kyma")
+		Eventually(removeModuleFromKyma, Timeout, Interval).
+			WithArguments(runtimeClient, remoteKyma.Name, remoteKyma.Namespace, moduleInKCP.Name).
+			Should(Succeed())
+
+		By("KCP Manifest is being deleted")
```
Comment (ruanxin) [inline controllers/control-plane/kyma_remote_sync_test.go:280]: Please add a test case to check KCP Kyma still contains those initiated modules in spec, We need to make sure those init modules do not get removed.

The follow up test case can be
1. delete SKR Kyma CR
2. verify all KCP Manifests created again

Comment (jeremyharisch) [reply controllers/control-plane/kyma_remote_sync_test.go:282]: Done

File: controllers/control-plane/kyma_remote_sync_test.go [hunk 2]
```diff
@@ -243,6 +235,76 @@ var _ = Describe("Kyma sync into Remote Cluster", Ordered, func() {
...
+		By("Remote Kyma contains default module")
+		Eventually(containsModuleInSpec, Timeout, Interval).
+			WithArguments(runtimeClient, remoteKyma.Name, remoteKyma.Namespace, moduleInKCP.Name).
+			Should(Succeed())
+
+		By("KCP Manifest is being created")
+		Eventually(ManifestExists, Timeout, Interval).
+			WithArguments(ctx, kyma, moduleInKCP, controlPlaneClient).
+			Should(Succeed())
+		By("KCP Manifest CR becomes ready")
```
Comment (ruanxin) [inline controllers/control-plane/kyma_remote_sync_test.go:270]: This step is not needed, if you need to update manifest CR manually, normally afterwards you check Kyma state become ready. But anyway in this test, it's not necessary to do this manual update.

Comment (jeremyharisch) [reply controllers/control-plane/kyma_remote_sync_test.go:270]: U r correct

**Review by NHingerl — APPROVED** [review]

**Pull request merged**